engine/language-server/src/main/scala/org/enso/languageserver/vcsmanager/Git.scala

```diff
@@ -0,0 +1,180 @@
+package org.enso.languageserver.vcsmanager
+
+import java.io.FileNotFoundException
+import java.nio.file.Path
+import org.enso.languageserver.effect.BlockingIO
+import org.eclipse.jgit.api.{Git => JGit}
+import org.eclipse.jgit.api.ResetCommand.ResetType
+import org.eclipse.jgit.api.errors.RefAlreadyExistsException
+import org.eclipse.jgit.api.errors.RefNotFoundException
+import org.eclipse.jgit.errors.RepositoryNotFoundException
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import org.eclipse.jgit.revwalk.{RevCommit, RevWalk}
+import org.enso.languageserver.vcsmanager.Git.MasterRef
+
+import scala.jdk.CollectionConverters._
+import zio.blocking.effectBlocking
+
+import java.time.Instant
+import scala.annotation.tailrec
+
+class Git extends VcsApi[BlockingIO] {
+  override def init(root: Path): BlockingIO[VcsFailure, Unit] = {
+    effectBlocking {
+      val rootFile = root.toFile
+      if (!rootFile.exists()) {
+        throw new FileNotFoundException("unable to find project repo: " + root)
+      }
+      val initCmd =
+        JGit
+          .init()
+          .setDirectory(rootFile)
+          .setBare(false)
+      val jgit = initCmd.call()
+      val commitCmd =
+        jgit
+          .commit()
+          .setAllowEmpty(true)
+          .setAll(true)
+          .setMessage("Initial commit")
+          .setAuthor("Enso VCS", "vcs@enso.io")
+      commitCmd.call()
+      ()
+    }.mapError(errorHandling)
+  }
+
+  private def repository(path: Path): Repository = {
+    val builder = new FileRepositoryBuilder();
+    builder
+      .setGitDir(path.resolve(Git.DefaultGitRepoDir).toFile)
+      .setMustExist(true)
+      .build()
+  }
+
+  override def commit(
+    root: Path,
+    named: Option[String]
+  ): BlockingIO[VcsFailure, Unit] = {
+    effectBlocking {
+      val repo = repository(root)
+      val jgit = new JGit(repo)
+      val addCmd =
+        jgit
+          .add()
+          .addFilepattern(".")
+
+      addCmd.call()
+
+      // Ensure that the name is unique
+      val commitName = named match {
```

Comment:
I'm thinking, instead of requiring a unique commit name, maybe we can use SHA checksums to manage commits? It should be more straightforward than searching for the name every time.

Reply:
Good idea. Ideally I would use tags, but then tags have restricted values